### PR TITLE
Rename pdata.AnyValueArray to AttributeSlice

### PR DESCRIPTION
**Pull request by Aneurysm9**
Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

### CODE REVIEW

File: model/pdata/common.go
Comment (bogdandrutu) [inline model/pdata/common.go:205]: SliceVal? I want consistency, maybe AttributeArray for the struct? Not sure what is best but I want consistency 👯

**Review by bogdandrutu — COMMENTED** [review]
LGTM in general except that small inconsistency, can be done in a separate PR but we need to confirm the names here.

**Review by carlosalberto — APPROVED** [review]

File: model/pdata/common.go
Comment (Aneurysm9) [reply model/pdata/common.go]: Changed to `SliceVal()` and also fixed a couple comments still referring to `AnyValueArray`.

File: model/pdata/common.go
Comment (codeboten) [inline model/pdata/common.go:200]: Nit
```suggestion
// SliceVal returns the slice value associated with this AttributeValue.
```

File: model/pdata/common.go
Comment (codeboten) [inline model/pdata/common.go:265]: Should this be `SetSliceVal`?

File: CHANGELOG.md
Comment (codeboten) [inline CHANGELOG.md:8]: Please add a note about ArrayVal becoming SliceVal as well.

**Review by codeboten — APPROVED** [review]
Thanks for addressing my comments!

**Review by bogdandrutu — APPROVED** [review]
Will not merge until contrib is green with what is merged in main. It means someone needs to upgrade contrib to pass with the current main before this gets merged to not have more than one breaking change to fix.

**Comment by tigrannajaryan** [comment]
Since the PR is fixing names I feel compelled to comment on naming despite 3 approvals :-)

AttributeSlice doesn't seem like a good name for an array of _values_. This is not a slice of _attributes_. Attributes are key/value _pairs_. This is a slice of AnyValue. May be call it AttributeValueSlice? (a bit too long IMO).

I believe attribute means a distinct thing and this uses the term incorrectly.

**Comment by bogdandrutu** [comment]
@tigrannajaryan sounds good to me `AttributeValueSlice` what about the other funcs on AttributeValue? is ok to use `SliceVal`?

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan sounds good to me `AttributeValueSlice` what about the other funcs on AttributeValue? is ok to use `SliceVal`?

Yes, I think it is fine to rename Array->Slice.

**Comment by Aneurysm9** [comment]
What then of `AttributeMap`?  It stores `*[]otlpcommon.KeyValue` whose values are `AnyValue`.

**Comment by bogdandrutu** [comment]
AttributeMap stores attributes, every entry has a "key" and a "AnyValue"

**Comment by bogdandrutu** [comment]
So I think the mapping is:
*`otlpcommon.KeyValue` => Attribute 
* `otlpcommon.AnyValue` => AttributeValue

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
